src/App.tsx

```diff
@@ -1,16 +1,23 @@
+import { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
 
 export const App = () => {
+  const [movieList, setMovieList] = useState([...moviesFromServer]);
+
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={movieList} />
       </div>
       <div className="sidebar">
-        <NewMovie /* onAdd={(movie) => {}} */ />
+        <NewMovie
+          onAdd={newMovie => {
+            setMovieList(movies => [...movies, newMovie]);
+          }}
```

Comment:
move the props you pass to an own function `handleAddMovie`
```suggestion
          onAdd={handleAddMovie}
```